samples/client/petstore/rust-reqwest/src/apis/configuration.rs

```diff
@@ -0,0 +1,67 @@
+/* 
+ * OpenAPI Petstore
+ *
+ * This is a sample server Petstore server. For this sample, you can use the api key `special-key` to test the authorization filters.
+ *
+ * OpenAPI spec version: 1.0.0
+ * 
+ * Generated by: https://openapi-generator.tech
+ */
+
+use std;
+
+use hyper;
+use reqwest;
+use reqwest::header::{Formatter, Header, Raw};
+
+pub struct Configuration {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub client: reqwest::Client,
+    pub basic_auth: Option<BasicAuth>,
+    pub oauth_access_token: Option<String>,
+    pub api_key: Option<String>,
+    // TODO: take an oauth2 token source, similar to the go one
+}
+
+pub type BasicAuth = (String, Option<String>);
+
+impl Configuration {
+    pub fn new() -> Configuration {
+        Configuration {
+            base_path: "http://petstore.swagger.io/v2".to_owned(),
+            user_agent: Some("Swagger-Codegen/1.0.0/rust".to_owned()),
+            client: reqwest::Client::new(),
+            basic_auth: None,
+            oauth_access_token: None,
+            api_key: None,
+        }
+    }
+
+    pub fn header_configured_api_key(&self) -> HeaderApiKey {
+        HeaderApiKey(self.api_key.clone().expect("API key not configured"))
+    }
```

Comment:
Why do we need this function? I'm a little nervous that the `expect` makes this somewhat unsafe to call.